math/minuit2/inc/Minuit2/SinParameterTransformation.h

```diff
@@ -28,9 +28,9 @@ class SinParameterTransformation {
 
    ~SinParameterTransformation() {}
 
-   double Int2ext(double Value, double Upper, double Lower) const;
-   double Ext2int(double Value, double Upper, double Lower, const MnMachinePrecision &) const;
-   double DInt2Ext(double Value, double Upper, double Lower) const;
+   long double Int2ext(long double Value, long double Upper, long double Lower) const;
+   long double Ext2int(long double Value, long double Upper, long double Lower, const MnMachinePrecision &) const;
+   long double DInt2Ext(long double Value, long double Upper, long double Lower) const;
```

Comment:
Should these be additional overloads, leaving the perf of the `double` version? Or have them as templates, only offering `double` and `long double` specializations (and possibly `float`)?

Reply:
We discussed this, and concluded that it's unlikely to cause issues. The two potential problems would be:

1. On some more exotic platform the long doubles may not be implemented or differently and the (bit-wise exact) tests will not pass. However, it is unclear what platforms this would be. GPU computation, for instance, would not be affected, because the parameter transformations are not part of the main calculation.
2. This also means performance is not expected to be a big issue. With big fits, the parameter transformations are a minute part of the Minuit run time. Most time by far is spent in the function calls. The only situation in which it could be an issue is if you have a very cheap function with a huge amount of parameters. This seems quite unlikely for realistic fits.

Reply:
One could also argue that a pro of using `long double` is that it clearly signals that these trigonometric transformations can lose some bits. This happens when you use the `double` versions and transform back and forth. In some situations, you can then lose 1 or 2 bits, just because of the trigonometric implementations. An alternative solution would be more precise sine etc functions.

Reply:
Thanks!

> and the (bit-wise exact) tests will not pass

That's curious - how do you "survive" on Windows? https://docs.microsoft.com/en-us/cpp/cpp/fundamental-types-cpp?view=msvc-160#floating-point-types

Reply:
Windows will use the double version and it will then use few bits when using the external calculator with the multiprocess. Will the multiprocess be available on Windows ? Probably not

Reply:
OK - but if there's only a `long double` version then which "double version" are you referring to? "Bit-wise exact" still depends on implementation / architecture. I trust this is handled, just making sure you're aware. (And I still don't know where these tests are.)

Reply:
The tests in #8596 for instance @Axel-Naumann :) Btw, they don't pass currently either, due to changes in master that I hadn't incorporated yet. On the todo list.

But good question about Windows. I haven't tested at all there yet. So probably, at least for the time being, it will have to be disabled on Windows, as @lmoneta says. In principle, ZeroMQ works on all platforms, so that is not the bottleneck. However, I have no idea how/whether forking works on Windows.

Reply:
Just so you don't have to dig into the code: the test compares the results of fitting with RooMinimizerFcn vs fitting the same function with RooGradMinimizerFcn. We put a lot of effort into making those two fits run *exactly* the same way. Small bit-wise differences quickly accumulate in the algorithm to end up with different results (past the desired precision, so e.g. in the 6th decimal if you wanted the fit to converge to 5).